File: Course-Registration-System-main/java-(iteration_1)/src/Lecturer.java

public class Lecturer extends Staff {
    private String title;


    public Lecturer(int id, String password, String name, String surname, String title) {
        super(id, password, name, surname);
        this.title=title;

    }

    // Getters and Setters for attributes
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }



    @Override
    public String getInfo() {
        // Implementation for viewing general lecturer information
        return("Lecturer:" + getTitle() + " " + getName() + " " + getSurname());
    }



}

File: Course-Registration-System-main/python-(iteration_2)/src/Lecturer.py
from Staff import Staff

class Lecturer(Staff):
    def __init__(self, id, password, name, surname, title):
        super().__init__(id, password, name, surname)
        self.__title = title

    # Getters and Setters for attributes
    def getTitle(self):
        return self.__title

    def setTitle(self, title):
        self.__title = title

    def getInfo(self):
        # Implementation for viewing general lecturer information
       return f"Lecturer: {self.getTitle()} {self.getName()} {self.getSurname()}"
